lib/marked.js

```diff
@@ -1094,7 +1094,8 @@ function escape(html, encode) {
 }
 
 function unescape(html) {
-  return html.replace(/&([#\w]+);/g, function(_, n) {
+	// explicitly match decimal, hex, and named HTML entities 
+  return html.replace(/&(#(?:\d+)|(?:#x[0-9A-Fa-f]+)|(\w+))/g, function(_, n) {
```

Comment:
I posted a PR matt-/marked#1 for your xss_html_entities branch with that little change but with no tests yet. Let me know how you think the tests should look like so I'll add some. Thanks.